certinel.go

```diff
@@ -65,15 +64,6 @@ func (c *Certinel) Close() error {
 // can be passed as the GetCertificate member in a tls.Config object. It is
 // safe to call across multiple goroutines.
 func (c *Certinel) GetCertificate(clientHello *tls.ClientHelloInfo) (*tls.Certificate, error) {
-	c.mu.RLock()
-	defer c.mu.RUnlock()
-
-	return c.certificate, nil
-}
-
-func (c *Certinel) loadCertificate(certificate tls.Certificate) {
-	c.mu.Lock()
-	defer c.mu.Unlock()
-
-	c.certificate = &certificate
+	cert, _ := c.certificate.Load().(*tls.Certificate)
```

Comment:
@rlguarino The `_, ok` type assertion form is used here because [`Load`](https://golang.org/pkg/sync/atomic/#Value.Load) returns an untyped `nil` before the first [`Store`](https://golang.org/pkg/sync/atomic/#Value.Store) has occurred – you can see that in [src/sync/atomic/value.go](https://github.com/golang/go/blob/c81a353/src/sync/atomic/value.go#L34-L37). After the first call to `Store`, a typed value will always be returned. By using the `_, ok` form, a typed nil will be returned rather than causing a panic.

While it is possible to do something else, this is exactly consistent with current behaviour. Before the first call to `loadCertificate`, `GetCertificate` already returns `(nil, nil)`. Based on the `crypto/tls` behaviour of falling back to `NameToCertificate` and `Certificates`, this seems correct to me.